src/uhs/transaction/validation.cpp

```diff
@@ -72,6 +77,38 @@ namespace cbdc::transaction::validation {
         return std::nullopt;
     }
 
+    auto
+    check_mint_tx(const cbdc::transaction::full_tx& tx,
+                  const std::unordered_set<pubkey_t, hashing::null>& minters)
+        -> std::optional<tx_error> {
+        // ensure there are outputs
```

Comment:
Let's `assert(tx.m_inputs.empty())` at the start of this function so we don't misuse it accidentally.